src/muc_light/mod_muc_light_codec_legacy.erl

```diff
@@ -56,15 +56,13 @@ encode({#msg{} = Msg, AffUsers}, Sender, {RoomU, RoomS} = RoomUS, HandleFun) ->
              {<<"from">>, RoomBin}
             ],
     MsgForArch = #xmlel{ name = <<"message">>, attrs = Attrs, children = Msg#msg.children },
-    EventData = [{from_nick, FromNick},
-                 {from_jid, Sender},
-                 {room_jid, jid:make_noprep({RoomU, RoomS, <<>>})},
-                 {affiliation, Aff},
-                 {role, mod_muc_light_utils:light_aff_to_muc_role(Aff)}
-    ],
-    FilteredPacket = #xmlel{ children = Children }
+    EventData = #{from_nick => FromNick,
+                  from_jid => Sender,
+                  room_jid => jid:make_noprep({RoomU, RoomS, <<>>}),
```

Comment:
This could perfectly be
```suggestion
                  room_jid => jid:make_noprep(RoomU, RoomS, <<>>),
```
without the tupling, if we're already changing it.